lib/glimesh/channel_hosts_lookups.ex

```diff
@@ -0,0 +1,138 @@
+defmodule Glimesh.ChannelHostsLookups do
+  @moduledoc """
+  Channel Hosting lookups
+  """
+
+  import Ecto.Query, warn: false
+
+  alias Glimesh.ChannelLookups
+  alias Glimesh.Repo
+  alias Glimesh.Streams.ChannelHosts
+
+  def get_channel_hosting_list(channel_id) do
+    ChannelHosts
+    |> where([ch], ch.hosting_channel_id == ^channel_id)
+    |> preload(host: [:user], target: [:user])
+    |> Repo.all()
+  end
+
+  def get_current_hosting_target(channel) do
+    ChannelHosts
+    |> where([ch], ch.hosting_channel_id == ^channel.id and ch.status == "hosting")
+    |> limit(1)
+    |> preload(host: [:user], target: [:user])
+    |> Repo.one()
+  end
+
+  def get_targets_host_info(host_username, target_channel) do
+    host = ChannelLookups.get_channel_for_username(host_username)
+
+    ChannelHosts
+    |> where(
+      [ch],
+      ch.hosting_channel_id == ^host.id and ch.status == "hosting" and
+        ch.target_channel_id == ^target_channel.id
+    )
+    |> preload(host: [:user], target: [:user])
+    |> Repo.one()
+  end
+
```

Comment:
I've used raw sql in the following methods for clarity purposes.  The Ecto.Query format was getting hard to read.